dialogue-annotations-processor/src/test/java/com/palantir/myservice/service/MyService.java

```diff
@@ -47,6 +48,12 @@ String greet(
     @Request(method = HttpMethod.GET, path = "/greeting", accept = CustomStringDeserializer.class)
     ListenableFuture<String> getGreetingAsync();
 
+    @Request(method = HttpMethod.GET, path = "/input-stream")
+    InputStream inputStream();
+
+    @Request(method = HttpMethod.GET, path = "/input-stream-custom", accept = CustomInputStreamDeserializer.class)
+    InputStream customInputStream();
```

Comment:
I think we're missing some test coverage for these, I don't believe the ErrorHandlingDeserializer will do what we want because unlike `ErrorHandlingDeserializerFactory`, the `ErrorHandlingDeserializer` closes responses in all cases before returning the result from the delegate deserializer.

Reply:
Indeed - I had pattern-matched from ErrorHandlingVoidSerializer but realize that this should behave differently. I'm not quite sure how I'd actually test the behavior here though. Do you have recommendations?

Reply:
Testing shouldn't be too bad, using the generated client and attempting to read a byte from the returned InputStream would be sufficient. We expect to read a byte, but if the read fails because the stream is closed, we know we have a bug.

> Also, I'd be inclined to remove `ErrorHandlingVoidDeserializer`

I don't think we can do this because `ErrorHandlingVoidDeserializer` is used in existing generated code, removal would cause runtime failures on transitive dependencies!

Reply:
Oh good catch on the ErrorHandlingVoidDeserializer! That's a bit sad 😅 

I'll see if I can implement a test 🤞